tecken/tests/test_upload.py

```diff
@@ -1268,119 +754,39 @@ def test_upload_duplicate_files_in_zip_different_name(fakeuser, client):
         assert response.json()["error"] == error_msg
 
 
-@pytest.mark.django_db
-def test_upload_client_unrecognized_bucket(fakeuser, client):
+def test_upload_client_unrecognized_bucket(client, db, s3_helper, uploaderuser):
     """The upload view raises an error if you try to upload into a bucket
     that doesn't exist."""
-    token = Token.objects.create(user=fakeuser)
+    token = Token.objects.create(user=uploaderuser)
     (permission,) = Permission.objects.filter(codename="upload_symbols")
     token.permissions.add(permission)
     url = reverse("upload:upload_archive")
 
-    with open(ZIP_FILE, "rb") as f, mock.patch(
-        "tecken.storage.StorageBucket.exists", return_value=False
-    ), pytest.raises(ImproperlyConfigured):
-        client.post(url, {"file.zip": f}, HTTP_AUTH_TOKEN=token.key)
+    with open(ZIP_FILE, "rb") as fp, pytest.raises(ImproperlyConfigured):
+        client.post(url, {"file.zip": fp}, HTTP_AUTH_TOKEN=token.key)
 
 
-def test_get_bucket_info(settings):
-    user = FakeUser("peterbe@example.com")
-
-    settings.UPLOAD_DEFAULT_URL = "https://s3.amazonaws.com/some-bucket"
-    bucket_info = get_bucket_info(user)
+def test_get_bucket_info(settings, uploaderuser):
+    settings.UPLOAD_DEFAULT_URL = "http://s3.amazonaws.com/some-bucket"
+    bucket_info = get_bucket_info(uploaderuser)
     assert bucket_info.name == "some-bucket"
     assert bucket_info.endpoint_url is None
     assert bucket_info.region is None
     assert not bucket_info.try_symbols
 
-    settings.UPLOAD_DEFAULT_URL = "https://s3-eu-west-2.amazonaws.com/some-bucket"
-    bucket_info = get_bucket_info(user)
+    settings.UPLOAD_DEFAULT_URL = "http://s3-eu-west-2.amazonaws.com/some-bucket"
+    bucket_info = get_bucket_info(uploaderuser)
     assert bucket_info.name == "some-bucket"
     assert bucket_info.endpoint_url is None
     assert bucket_info.region == "eu-west-2"
 
     settings.UPLOAD_DEFAULT_URL = "http://s3.example.com/buck/prefix"
-    bucket_info = get_bucket_info(user)
+    bucket_info = get_bucket_info(uploaderuser)
     assert bucket_info.name == "buck"
     assert bucket_info.endpoint_url == "http://s3.example.com"
     assert bucket_info.region is None
 
 
-def test_get_bucket_info_try_symbols(settings):
```

Comment:
This test makes no sense to me. Users are put into groups and there's a "Uploaders" group which has both permissions. So this situation isn't possible and therefore we shouldn't be testing it.